### ISSUE TITLE
bug in nano_bcn_uart_init function

### ISSUE THREAD

**Issue by manuelboebel** [NanoBeacon/IN100-Host-Controller-Sample#1]
There's a bug in the nano_bcn_uart_init function (nano_bcn_serial_protocol.c / line 341): 
It checks twice, if p_hif->delay is not NULL, but does not check p_hif->serial_break.

```
int nano_bcn_uart_init(host_itf_t *p_hif)
{
	if (!p_hif->delay)
		return NANO_BCN_ERR_INVALID_PARAM;

	if (!p_hif->delay) 
		return NANO_BCN_ERR_INVALID_PARAM;

	if (!p_hif->serial_rx) 
		return NANO_BCN_ERR_INVALID_PARAM;

	if (!p_hif->serial_tx) 
		return NANO_BCN_ERR_INVALID_PARAM;
...
}
```

**Comment by wenzhengliu2022** [comment]
Some MCU' serial ports do not support break.

**Comment by manuelboebel** [comment]
Code for checking, if serial break function reference is available has been added in commit bdf06a73b054bac1885e37b93da1d5dac8ed1156. Thanks @wenzhengliu2022!
